crates/cli/src/main.rs

```diff
@@ -334,6 +339,17 @@ mod linux {
                 .arg(ipc_url)
                 .status()
         }
+
+        async fn get_env_vars(&self) -> Option<HashMap<String, String>> {
+            // The desktop entry uses `cli` to spawn `zed`, so we need to load env vars from the shell
+            // since it doesn't inherit env vars from the terminal.
+            if !std::io::stdout().is_terminal() {
+                load_shell_from_passwd().await.log_err();
+                load_login_shell_environment().await.log_err();
```

Comment:
* Looking at 

https://github.com/zed-industries/zed/blob/099bdcd94cf9c12d010790ed0dffc7684317d2f0/crates/zed/src/main.rs#L217-L228

it seems to call `load_login_shell_environment` for every OS, but here we do it for Linux and FreeBSD only, is it correct to omit macOS here?

* Same question for `load_shell_from_passwd`, `#[cfg(unix)] ` does include macOS.

Reply:
Yes, this is intentional. I'm not familiar with macOS, but as far as I know, there aren't any cases where the `cli` binary would be run outside of a terminal environment, such as through a `.desktop` file in Linux. I assume the macOS app menu uses the `zed` binary instead of the `cli` binary. If the `cli` binary were used, the original issue would likely occur on macOS as well. Unfortunately, I don't have a macOS device to verify this.

Reply:
Makes sense, thank you for the explanations, macOS does run the editor binary when spawned out of the Dock.